client/lib/feed-url-store/actions.js

```diff
@@ -0,0 +1,24 @@
+var inflight = require( 'lib/inflight' ),
+	wpcom = require( 'lib/wp' );
+
+function requestKey( feedId ) {
+	return `feed-${feedId}`;
+}
+
+const FeedUrlStoreActions = {
+	discover: function( feedUrl, fn ) {
+		const key = requestKey( feedUrl );
+
+		if ( inflight.requestInflight( key ) ) {
+			return;
```

Comment:
this should really do three things: 
- look to see if we already have a mapping, and return Promise.resolve( theMapping)
- if not, see if we have a request inflight (just stash a map of url -> promise). if we do, return the existing promise. if not, 
- make a new request and stash the Promise in the map of url -> promise